runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -202,6 +204,12 @@ public String registerEnvironment(Environment env) {
     return name;
   }
 
+  public String getOnlyEnvironmentId() {
+    // TODO Support sumtiple environments. The environment should be decided by the traslation.
```

Comment:
* typo: multiple
* Why do you need to `ImmutableSet.copyOf()` - doesn't Iterables.getOnly work on keySet() as-is?
* Can we standartize the key in this map, so that the default java environment always has the same id?

Reply:
* will fix the typo.
* Yup, we don't need a copy of keyset anymore.
* The key has to be the URL of docker image so it has to be dynamic. Sorry, I don't understand what standardization you mean.

Reply:
Oh I thought the key was just some sort of unique id, and the URL is inside the Environment proto itself. Is it both?..

Reply:
Yup, based on the implementation, key is the URL. 
Environment only have a single filed URL so the additional book keeping is redundant for now.

Reply:
Actually the key is not necessarily a URL. SdkComponents.registerEnvironment() currently takes an Environment and generates an id for it *based* on its URL, but there's no code that *expects* the id to be a URL. We can use any unique string here. I think it may make sense to change SdkComponents to put the Java environment under some standardized ID, e.g. "javaDefault" or something.